### DISCUSSION TITLE
Ability for `magit-diff-visit-file` to take a custom buffer-setting function

### DISCUSSION THREAD

**Discussion by vemv** [magit/magit#5111] [Support] [answered]
Hi!

I'd need `magit-diff-visit-file` to be able to be customized with a function that will decide where to render the buffer.

As far as I know, `switch-to-buffer-other-window`, `switch-to-buffer-other-frame` can be a bit arbitrary in how they select a window/frame.

(I am aware that there's `display-buffer` customization, but that is a global option. I only want to customize buffer rendering when invoked from this particular Magit feature).

A `(lambda (buffer)` that I can set as a defcustom would suffice.

Thanks - V

**Comment by tarsius** [comment, marked as answer]
(I agree that Magit, and Emacs in general, don't always display buffers optimally. I also intend to work on "fixing" that eventually.)

The `magit-diff-visit-file` family of commands is already easily customizable--it's just not through a custom option. Create one or more new commands that use `magit-diff-visit-file--internal`, and grep for `#'magit-diff-visit-`, to find bindings you might want to change.

**Reply by vemv** [reply]
Thanks!

Yes, I figured that I could use `magit-diff-visit-file--internal` although I generally try not to use double-dashed functions from third party libs.

Although given your indication, I take it that I can trust that primitive not to suddenly go away.

**Reply by tarsius** [reply]
At some point this *is* going to break, but that will be a huge change, which you surely will hear about, and if I added an option instead, that option would stop functioning at the same time too.
